sdk/resource/resource.go

```diff
@@ -17,95 +17,22 @@
 package resource
 
 import (
-	"sort"
-	"strings"
-
 	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/label"
 )
 
-// Resource describes an entity about which identifying information and metadata is exposed.
 type Resource struct {
-	sorted []core.KeyValue
-	keySet map[core.Key]struct{}
+	labels label.Set
 }
 
 // New creates a resource from a set of attributes.
 // If there are duplicates keys then the first value of the key is preserved.
 func New(kvs ...core.KeyValue) *Resource {
-	res := &Resource{keySet: make(map[core.Key]struct{})}
-	for _, kv := range kvs {
-		// First key wins.
-		if _, ok := res.keySet[kv.Key]; !ok {
-			res.keySet[kv.Key] = struct{}{}
-			res.sorted = append(res.sorted, kv)
-		}
+	return &Resource{
+		labels: label.NewSet(kvs...),
 	}
-	sort.Slice(res.sorted, func(i, j int) bool {
-		return res.sorted[i].Key < res.sorted[j].Key
-	})
-	return res
-}
-
-// String implements the Stringer interface and provides a reproducibly
-// hashable representation of a Resource.
-func (r Resource) String() string {
-	// Ensure unique strings if key/value contains '=', ',', or '\'.
-	escaper := strings.NewReplacer("=", `\=`, ",", `\,`, `\`, `\\`)
-
-	var b strings.Builder
-	// Note: this could be further optimized by precomputing the size of
-	// the resulting buffer and adding a call to b.Grow
-	b.WriteString("Resource(")
-	if len(r.sorted) > 0 {
-		b.WriteString(escaper.Replace(string(r.sorted[0].Key)))
-		b.WriteRune('=')
-		b.WriteString(escaper.Replace(r.sorted[0].Value.Emit()))
-		for _, s := range r.sorted[1:] {
-			b.WriteRune(',')
-			b.WriteString(escaper.Replace(string(s.Key)))
-			b.WriteRune('=')
-			b.WriteString(escaper.Replace(s.Value.Emit()))
-		}
-
-	}
-	b.WriteRune(')')
-
-	return b.String()
-}
-
-// Attributes returns a copy of attributes from the resource in a sorted order.
-func (r Resource) Attributes() []core.KeyValue {
```

Comment:
This rename will need to be propagated to all callers.

Reply:
I restored `Attributes`.